src/NServiceBus.Core.Tests/Serializers/XML/SerializerTests.cs

```diff
@@ -1227,10 +1227,78 @@ public void Object_property_with_primitive_or_struct_value_should_serialize_corr
             Assert.AreEqual(message.Value, ((SerializedPair)messageDeserialized[0]).Value);
         }
 
+
+        [Test]
+        public void Should_throw_exception_when_deserializing_payloads_with_types_existing_in_deployed_assemblies_but_not_pre_discovered_as_message_types()
+        {
+            var xmlWithBaseType = @"<?xml version=""1.0"" ?>
+                <Messages
+                    xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance""
+                    xmlns:xsd=""http://www.w3.org/2001/XMLSchema""
+                    xmlns=""http://tempuri.net/NServiceBus.Core.Tests.Serializers.XML""
+                    xmlns:baseType=""NServiceBus.Core.Tests.Serializers.XML.IMyBusMessage"">
+                    <FirstMessage></FirstMessage>
+                    <SecondMessage></SecondMessage>
+                    <SecondMessage></SecondMessage>
+                    <SecondMessage></SecondMessage>
+                </Messages>
+                ";
+
+            using (var stream = new MemoryStream())
+            {
+                var streamWriter = new StreamWriter(stream);
```

Comment:
Isn't streamwriter disposable too? Not that it really matters for the test but then you could it move on the same level like the MemStream. Also can we start using the new using syntax?

Reply:
I don't think we can use the new using syntax (c# 8) with .net472 as a target?

Reply:
Using is one of the features that works https://stackoverflow.com/questions/56651472/does-c-sharp-8-support-the-net-framework